extensions/interactions/NumberWithUnits/directives/oppia-interactive-number-with-units.component.ts

```diff
@@ -13,113 +13,134 @@
-        }).result.then(function() {}, function() {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-        });
-      };
-      ctrl.$onInit = function() {
-        $scope.$watch('$ctrl.answer', function(newValue) {
-          try {
-            NumberWithUnitsObjectFactory.fromRawInputString(newValue);
-            errorMessage = '';
-            ctrl.NumberWithUnitsForm.answer.$setValidity(
-              FORM_ERROR_TYPE, true);
-          } catch (parsingError) {
-            errorMessage = parsingError.message;
-            ctrl.NumberWithUnitsForm.answer.$setValidity(
-              FORM_ERROR_TYPE, false);
-          }
-        });
-        if (ctrl.savedSolution !== undefined) {
-          let savedSolution = ctrl.savedSolution;
-          savedSolution = NumberWithUnitsObjectFactory.fromDict(
-            savedSolution).toString();
-          ctrl.answer = savedSolution;
-        } else {
-          ctrl.answer = '';
-        }
-        ctrl.labelForFocusTarget = $attrs.labelForFocusTarget || null;
-
-        ctrl.NUMBER_WITH_UNITS_FORM_SCHEMA = {
-          type: 'unicode',
-          ui_config: {}
-        };
-
-        try {
-          NumberWithUnitsObjectFactory.createCurrencyUnits();
-        } catch (parsingError) {}
-
-        CurrentInteractionService.registerCurrentInteraction(
-          submitAnswerFn, ctrl.isAnswerValid);
-      };
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { InteractionRulesService } from 'pages/exploration-player-page/services/answer-classification.service';
+
+import { NumberWithUnitsAnswer, InteractionAnswer } from 'interactions/answer-defs';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { HelpModalNumberWithUnitsComponent } from './oppia-help-modal-number-with-units.component';
+import { NumberWithUnitsObjectFactory } from 'domain/objects/NumberWithUnitsObjectFactory';
+import { NumberWithUnitsRulesService } from './number-with-units-rules.service';
+
+@Component({
+  selector: 'oppia-interactive-number-with-units',
+  templateUrl: './number-with-units-interaction.component.html',
+  styleUrls: []
+})
+export class InteractiveNumberWithUnitsComponent
+    implements OnInit, OnDestroy {
+  @Input() labelForFocusTarget: string;
+  @Input() savedSolution: InteractionAnswer;
+  componentSubscriptions: Subscription = new Subscription();
+  FORM_ERROR_TYPE: string = 'NUMBER_WITH_UNITS_FORMAT_ERROR';
+  errorMessage: string = '';
+  answer: string = '';
+  isValid: boolean = true;
```

Comment:
Why these default values?

Reply:
Done, may be add while testing in local machine, now removed. NO NEED.